gateway-shell/src/main/java/org/apache/knox/gateway/shell/table/KnoxShellTable.java

```diff
@@ -76,6 +80,98 @@ public KnoxShellTable value(Comparable<? extends Object> value) {
     return col;
   }
 
+  public double[] toDoubleArray(String colName) throws IllegalArgumentException {
+    List<Comparable<? extends Object>> col = values(colName);
+    double[] colArray = new double[col.size()];
+    int conversionMethod = 0;
+    for (int i = 0; i < col.size(); i++) {
+      if (i == 0) {
+        if (col.get(i) instanceof Integer) {
+          conversionMethod = INTEGER_CONVERSION_METHOD;
```

Comment:
Do we need to do this check and then later do a complicated convert to `double`? Is it possible to just do something like `Double.valueOf(col.get(i))`?

Reply:
https://repl.it/repls/GigaLavenderInternalcommand

Shows that it looks like this works just using `Double.valueOf(...)`

Reply:
The check is needed as `valueOf` can't be called on a Comparable. You would need to convert to something like string before doing that. I had something similar to this originally (my comment on the jira) where it was `Double.parseDouble(col.get(i).toString());`, which is not efficient. With this though we're just going from whichever underlying datatype it happens to be out to double.